scipy/special/tests/test_binomln.py

```diff
@@ -0,0 +1,152 @@
+import numpy as np
+import pytest
+
+from numpy.testing import assert_allclose, assert_equal
+from scipy.special._ufuncs import _binomln
+
+
+@pytest.mark.parametrize(
+    # # Test cases generated for various signs and orders of magnitude of
+    # # n and k using the following script:
+
+    # import numpy as np
+
+    # from mpmath import mp
+    # from scipy.special._special_ufuncs import _binomln
+
+
+    # def mp_binomln(n, k):
+    #     with mp.workdps(400):
+    #         result = float(mp.log(abs(mp.binomial(n, k))))
+    #     return result
+
+
+    # rng = np.random.default_rng(204054655279796806347416808029333184006)
+
+    # cases = np.asarray([
+    #     [rng.uniform(3, 10), rng.uniform(2, 3)],
+    #     [rng.uniform(3, 10), rng.uniform(1, 10) * 1e-15],
+    #     [rng.uniform(3, 10), rng.uniform(1, 10) * 1e-30],
+    #     [rng.uniform(1, 10)*1e-10, rng.uniform(2, 3)],
+    #     [rng.uniform(1, 10)*1e-10, rng.uniform(1, 10)*1e-8],
+    #     [rng.uniform(1, 10) * 10, rng.uniform(1, 10)],
+    #     [rng.uniform(1, 10) * 1e2, rng.uniform(1, 10) * 10],
+    #     [rng.uniform(1, 10) * 1e2, rng.uniform(1, 10)],
+    #     [rng.uniform(1, 10) * 1e3, rng.uniform(1, 10) * 1e-40],
+    #     [rng.uniform(1, 10) * 1e3, rng.uniform(1, 10) * 100],
+    #     [rng.uniform(1, 10) * 1e3, rng.uniform(1, 10) * 10],
+    #     [rng.uniform(1, 10) * 1e3, rng.uniform(1, 10)],
+    #     [rng.uniform(1, 10) * 1e3, rng.uniform(1, 10) * 1e-40],
+    #     [rng.uniform(1, 10) * 1e6, rng.uniform(1, 10) * 1e5],
+    #     [rng.uniform(1, 10) * 1e6, rng.uniform(1, 10) * 1e3],
+    #     [rng.uniform(1, 10) * 1e6, rng.uniform(1, 10) * 10],
+    #     [rng.uniform(1, 10) * 1e6, rng.uniform(1, 10) * 1e-20],
+    # ])
+
+    # cases = np.vstack(
+    #     [
+    #         cases,
+    #         cases * np.asarray([[-1, 1]]),
+    #         cases * np.asarray([[1, -1]]),
+    #         cases * np.asarray([[-1, -1]]),
+    #     ]
+    # )
+
+    # test_cases = []
+    # for n, k in cases:
+    #     desired = mp_binomln(n, k)
+    #     actual = _binomln(n, k)
+    #     relative_error = abs(observed_result - expected_result) / abs(expected_result)
+    #     rtol = f"1e{int(max(-15, np.ceil(np.log10(relative_error))))}"
```

Comment:
I don't think these lines match the actual code used to generate this. One difference is the name of `desired`/`actual`, the other is that you wouldn't get `rtol=5e-15` with this code.

Reply:
Oh, good point. I changed the actual script but not the comment. Also the script in the comment is a nonworking version while I was in the middle of updating things. The choice of test cases is probably not that good. I threw this together quickly, the intent was just to get something that looked reasonable without thinking too much about it. And `rtol=5e-15` came from a manual bump to get that test to pass on all platforms in CI. We can definitely do better than this and use fewer magic numbers.